src/peft/tuners/ia3/layer.py

```diff
@@ -198,15 +198,15 @@ def __init__(
         IA3Layer.__init__(self, base_layer, is_feedforward=is_feedforward)
         self.fan_in_fan_out = fan_in_fan_out
         self._active_adapter = adapter_name
+        self._kernel_rank = base_layer.weight.dim()
```

Comment:
I was thinking that "dim" might be confusing, a "2D" convolution will in general have a 4D weight tensor (batch, channel, height, width) so I was trying to differentiate spatial dimension from tensor dimension. Although since I'm checking the "dim" attribute maybe it is better to be consistent.

Reply:
Personally, I find it less confusing than rank, but I'm open to other suggestions.